bootstrap.py

```diff
@@ -747,11 +738,31 @@ class MEOptions:
         print "\nExample usage: ./bootstrap.py -l admin -s foreman.example.com -o 'Default Organization' -L 'Default Location' -g My_Hostgroup -a My_Activation_Key"
         sys.exit(1)
 
+    # > Gather FQDN, HOSTNAME and DOMAIN using socket.getfqdn()
+    # > If socket.fqdn() returns an FQDN, derive HOSTNAME & DOMAIN using FQDN
+    # > else, HOSTNAME isn't an FQDN
+    # > if user passes --override-fqdn set FQDN, HOSTNAME and DOMAIN to the parameter that is given.
+    FQDN = socket.getfqdn()
+    if FQDN.find(".") != -1:
+        HOSTNAME = FQDN.split('.')[0]
+        DOMAIN = FQDN[FQDN.index('.') + 1:]
+    else:
+        HOSTNAME = FQDN
+        DOMAIN = None
+
+    if options.override_fqdn:
+        HOSTNAME = options.override_fqdn
+        FQDN = options.override_fqdn
+        DOMAIN = FQDN[FQDN.index('.') + 1:]
```

Comment:
this will crash if the user passed a non-fqdn to --override-fqdn (and they will)